Added new entry on Software Engineering (#2041)
Codecademy/docs: content/general/concepts/software-engineering/software-engineering.md

---
Title: 'Software Engineering'
Description: 'Software engineering is a field of study that involves using engineering principles to design, build, test, and maintain software systems.'
Subjects:

>   - 'Mobile Development'

Reviewer: ```suggestion
```

  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Development'
  - 'Servers'
  - 'Frameworks'
  - 'Developer Tools'
CatalogContent:
  - 'learn-how-to-code'
  - 'paths/full-stack-engineer-career-path'

>   - 'paths/back-end-engineer-career-path'

Reviewer: ```suggestion
```

Reviewer: We limit the subject and content fields to two items, tags can contain as many as five

## Software Engineering vs. Computer Science

Software engineering and computer science are closely related, and many software engineers have degrees or certifications in computer science. Both fields include fundamental information on computer science and programming. However, a key difference is that computer science focuses more on theoretical and mathematical aspects related to computing, and software engineering focuses more on practical aspects related to building software systems.

Software engineers apply their knowledge of computer science to address the practical problems related to software development and maintenance. Software engineering also involves implementing some of the many topics which come out of computer science.

Additionally, software engineers must perform other tasks that relate to the [software development life cycle](https://www.codecademy.com/resources/docs/general/software-development-life-cycle). This can include quality assurance, documentation, and working with clients or stakeholders.

## Types of Software Engineers

Software engineers are commonly categorized into three types: front-end, back-end, and full-stack.

### Front-End Engineers

> A [front-end](https://www.codecademy.com/resources/docs/general/front-end) engineer creates and manages the user-facing components of an application, such as a website. They typically work on web applications, although some front-end engineers work on desktop and mobile applications. Front-end engineers are familiar with languages such as JavaScript, [TypeScript](https://www.codecademy.com/resources/docs/typescript), [Ruby](https://www.codecademy.com/resources/docs/ruby), [JSON](https://www.codecademy.com/resources/docs/general/json), [HTML](https://www.codecademy.com/resources/docs/html), and [CSS](https://www.codecademy.com/resources/docs/css). They also use web development tools and frameworks, such as [Angular](https://www.codecademy.com/resources/docs/open-source/angular), [React](https://www.codecademy.com/resources/docs/react), and [Bootstrap](https://www.codecademy.com/resources/docs/open-source/bootstrap).

Reviewer: ```suggestion
A [front-end](https://www.codecademy.com/resources/docs/general/front-end) engineer creates and manages the user-facing components of an application, such as a website. They typically work on web applications, although some front-end engineers work on desktop and mobile applications. Front-end engineers are familiar with programming languages such as JavaScript, [TypeScript](https://www.codecademy.com/resources/docs/typescript), [Ruby](https://www.codecademy.com/resources/docs/ruby), as well as mark-up languages [HTML](https://www.codecademy.com/resources/docs/html), and [CSS](https://www.codecademy.com/resources/docs/css). They also use web development tools and frameworks, such as [Angular](https://www.codecademy.com/resources/docs/open-source/angular), [React](https://www.codecademy.com/resources/docs/react), and [Bootstrap](https://www.codecademy.com/resources/docs/open-source/bootstrap).
```
